### ISSUE TITLE
The pretrained model link in Google Drive is empty.

### ISSUE THREAD

**Issue by MaxChanger** [chenfengxu714/YOGO#2]
As the title says, the Google Drive folder is empty.

**Comment by chenfengxu714** [comment]
Thanks, we will release the pretrained model after 8/1/2021 due to some rights checking issues. BTW directly training from scratch would be fast if you were urgent to use. Feel free to contact me if you have any issues.

**Comment by MaxChanger** [comment]
Ok, I have trained on shapenet and got the result, mean_iou: 84.36, which is a little different from the paper.
In addition, the README does not clearly indicate the version used by S3DIS and how to preprocess it. According to the path in the config containing "pointcnn", I used the script in its repo to process it, but the dimension was wrong. Can you give some details

**Comment by MaxChanger** [comment]
In short, I guess it should be that YOGO needs 9-dimensional input with normal, but only xyz and rgb are obtained after pointcnn processing.

**Comment by chenfengxu714** [comment]
Thanks for indicating my missing data-preprocess code. I have updated it in readme now. You can go to data/s3dis folder and run python prepare_data.py to get the preprocessed s3dis data. Remember to change your data root. We use nine dimension inputs including x, y, z, r, g, b, and normalized x, y, z. 

As for the results, we report the evaluation results instead of training-then-evaluation results. During the training, the shown validation results are indeed slightly worse.  You can try to re-evaluate it by running our evaluation code.

**Comment by MaxChanger** [comment]
Thanks a lot for the detailed explanation of the relationship S3DIS, sorry I didn't notice data/s3dis/prepare_data.py. I am processing the data and believe it will work as input.

As for evaluation, I use the inference command in your README, and use the best.pth.tar after training. I want to know if I have misunderstood anything.

```shell
python train.py configs/shapenet/yogo/yogo.py --devices 2 --evaluate --configs.evaluate.best_checkpoint_path ./runs/shapenet.yogo.yogo/best.pth.tar
```

**Comment by chenfengxu714** [comment]
Yes, you are right to run this for evaluation.

**Comment by MaxChanger** [comment]
mean_iou: 84.36, which is exactly the result I got using this command.
I am currently training for the second time, ruling out that it is due to randomness, also I am using the environment of pytorch 1.7.0, cuda 11.1, RTX 3090.

**Comment by chenfengxu714** [comment]
I suppose you use our default setting vote number = 1 in the config file. Sorry for confusing you. As illustrated in the paper, we vote 10 times during the inference of ShapeNet, you can revise it into 10. (I have set 10 as the default setting now). Note that this is a very common evaluation in other methods, like PointNet++ votes 12 times, and RandLANet votes 100 times. Thanks.

**Comment by MaxChanger** [comment]
Thank you for the detailed answer, I will try it again.
By the way, S3DIS runs successfully according to your preprocessing code, thanks! 🍻

**Comment by chenfengxu714** [comment]
Also thank you for your feedback to help us improve the quality of this repo :). Also, feel free to contact me if you have other issues.
